net/advanced-operations/securing-and-signing/sign-pdf-document-from-smart-card/_index.md

```diff
@@ -110,20 +110,23 @@ using (FileStream fs = new FileStream(dataDir + "externalSignature1.pdf", FileMo
         };
 
         field1.PartialName = "sig1";
-        doc.Form.Add(field1, 1);
+        document.Form.Add(field1, 1);
         field1.Sign(externalSignature);
-        doc.Save();
+        document.Save();
     }
 }
 
-using (PdfFileSignature pdfSign = new PdfFileSignature(new Document(dataDir + "externalSignature1.pdf")))
+using (Document document = new Document(dataDir + "externalSignature1.pdf"))
 {
-    IList<string> sigNames = pdfSign.GetSignNames();
-    for (int index = 0; index <= sigNames.Count - 1; index++)
+    using (PdfFileSignature pdfSign = new PdfFileSignature(document))
     {
-        if (!pdfSign.VerifySigned(sigNames[index]) || !pdfSign.VerifySignature(sigNames[index]))
+        IList<string> sigNames = pdfSign.GetSignNames();
+        for (int index = 0; index <= sigNames.Count - 1; index++)
         {
-            throw new ApplicationException("Not verified");
+            if (!pdfSign.VerifySigned(sigNames[index]) || !pdfSign.VerifySignature(sigNames[index]))
```

Comment:
You should use  pdfSign.VerifySignature(sigNames[index]) only.  The VerifySigned is deprecated.
The VerifySigned is equivalent  of VerifySignature.